### PR TITLE
[AIDAPP-107]: Introduce configurable layout options for the KB portal

### PR DESCRIPTION
**Pull request by dgoetzit**
### Ticket(s) or GitHub Issue

- https://canyongbs.atlassian.net/browse/AIDAPP-107

### Technical Description

> Adds configurability for the portal layout through the admin management interface.

### Any deployment steps required?

> No

_______________________________________________

### CODE REVIEW

**Review by mxm1070 — CHANGES_REQUESTED** [review]

File: database/settings/2024_06_02_012958_add_portal_layout_to_portal_settings.php
[the remark this answers is not in the document]
Comment (Orrison) [reply database/settings/2024_06_02_012958_add_portal_layout_to_portal_settings.php:1]: Yeah, any data migrations should be separate from schema migrations. And feature flag toggling counts as such.

So let's split this into separate migrations.

**Review by Orrison — CHANGES_REQUESTED** [review]

**Review by mxm1070 — APPROVED** [review]

File: database/migrations/2024_06_03_200657_data_deactivate_portal_layout_feature_flag.php [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+</COPYRIGHT>
+*/
+
+use Laravel\Pennant\Feature;
+use Illuminate\Database\Migrations\Migration;
+
+return new class () extends Migration {
+    public function up(): void
+    {
+        Feature::purge('portal-configuration-options');
```
Comment (Orrison) [inline database/migrations/2024_06_03_200657_data_deactivate_portal_layout_feature_flag.php:43]: Why is this purge migration in this PR? Wouldn't this disable the feature?

Was this intended to be put in a cleanup branch?

**Review by Orrison — CHANGES_REQUESTED** [review]

File: database/migrations/2024_06_03_200657_data_deactivate_portal_layout_feature_flag.php (hunk 1 above)
Comment (danharrin) [reply database/migrations/2024_06_03_200657_data_deactivate_portal_layout_feature_flag.php:43]: Because you merged the cleanup PR into this one 😆

Comment (Orrison) [reply database/migrations/2024_06_03_200657_data_deactivate_portal_layout_feature_flag.php]: 😬

**Review by Orrison — APPROVED** [review]

**Pull request merged**